src/main.rs

```diff
@@ -40,6 +42,13 @@ fn determine_state_positions<D: DFA>(
                 state_vec.push(next_state);
             }
         }
+        current_index += 1;
+    }
+
+    state_vec.shuffle(&mut thread_rng());
```

Comment:
This should be `state_vec[1..].shuffle(&mut thread_rng());` since the init state should not be moved away from root